drizzlepac/hapsequencer.py

```diff
@@ -812,3 +825,222 @@ def _get_envvar_switch(envvar_name, default=None):
+            # from the current image
+            else:
+                grism_wcs_set &= set(dict_names.values())
+
+            log.info("***********8grism set: {}".format(grism_wcs_set))
+
+            # Oops...no common wcsnames
+            if not grism_wcs_set:
+                log.error("There are no common WCS solutions with this Grism/Prism image {} and previously processed images".format(filename))
+                log.error("    There is a problem with this Grism/Prism image/visit.")
+                sys.exit(1)
+        # If there are no WCS solutions, the image could be bad (e.g., EXPTIME=0 or EXPFLAG="TDF-DOWN...")
+        else:
+            log.warning("There are no WCS solutions in the Grism/Prism image {} in this visit.".format(filename))
+            log.warning("    Skip this image.")
+            skip_grism_list.append(filename)
+
+    log.info("WCS solutions common to all viable Grism/Prism images: {}".format(grism_wcs_set))
+
+    # There is a preference for the active WCS for the viable images in the visit
+    # Check the grism_wcs_set for the preferential solutions
+    # If a common name is found here, wait to report it as the active WCS may still need to
+    # be rolled back depending upon the WCS solutions available in the direct images.
+    match_list = []
+    for wcs_item in wcs_preference:
+        match_list = fnmatch.filter(grism_wcs_set, wcs_item)
+        if match_list:
+            grism_wcsname = match_list[0]
+            log.info("Proposed WCS for use after Grism/Prism examination: {}".format(grism_wcsname))
+            break
+
+    if not grism_wcsname:
+        log.error("None of the preferred WCS names are present in the common set of WCS names for the Grism/Prism images.")
+        log.error("    There is a problem with this visit")
+        sys.exit(1)
+
+        log.info("\nProposed WCS for use after Grism/Prism examination: {}".format(grism_wcsname))
+
+    # If it is now known there is at least one viable Grism/Prism observation, it is desired to make
+    # the active Grism/Prism WCS be the same for all of the Grism/Prism as well as the direct images
+    # in the visit.  The catch can be if any of the direct images do not have an 'a priori' solution
+    # which matches the Grism/Prism images.  In this case, it will be necessary to "fall back" to
+    # the pipeline WCS solution.
+    log.info("\n***** Direct Image Processing *****")
+    direct_wcsname = ''
+    direct_wcs_set = set()
+    skip_direct_list = []
+    exist_direct_set = False
+    # Loop over all the direct images for this detector in the visit
+    for edp in tdp.edp_list:
+        filename = edp.full_filename
+        hdu = fits.open(filename)
+
+        # Make sure the WCS is up-to-date - doing the update for the
+        # direct images here so there is no need to modify a
+        # pre-existing class which could cause an issue
+        drizcorr = hdu[0].header['DRIZCORR']
+        if drizcorr == "OMIT":
```

Comment:
It is unclear why this logic is included.  Why limit the WCSs to pipeline-only WCSs when you expect no other WCSs to have been added to the file already? If DRIZCORR was OMIT, we are almost guaranteed that 'updatewcs' has not been run on the data and no a priori or pipeline-default WCSs have been added to the file or SCI header.  The only way an a priori WCS solution will be appended to a direct image is using 'updatewcs(filename, use_db=True)'.  When 'use_db=False', you are restricting the WCSs being included in the SCI header to only the pipeline-default, unless the file has already been processed previously with 'updatewcs(filename, use_db=True)', something which is not guaranteed.  Even with 'use_db=True', the pipeline-default WCS will be added to the SCI header, even if it is just as an alternate WCS.

Reply:
Changed the if/else to just an if: if drizcorr is equal to OMIT, then updatewcs with use_db=True.